test/gtest/mhp/slide_view.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+#include <dr/mhp/views/sliding.hpp>
+
+template <typename T> class Slide : public testing::Test {};
+
+TYPED_TEST_SUITE(Slide, AllTypes);
+
+TYPED_TEST(Slide, is_compliant) {
+  TypeParam dv(10, dr::halo_bounds(2, 1, false));
+  LocalVec<TypeParam> lv(10);
+  iota(dv, 100);
+  std::iota(rng::begin(lv), rng::end(lv), 100);
+
+  auto local_sliding_view = rng::sliding_view(lv, 4); // halo_bounds + 1
+  auto dv_sliding_view [[maybe_unused]] = xhp::views::sliding(dv);
```

Comment:
I think there is a benefit to supporting a sliding api that takes a window size, even if passing anything other than the halo size is an error. My main motivation is that the view of the same name in ranges-v3 takes a window size. Secondary is for readability. I don't object to making the window size optional.

Reply:
If we are going to support ranges-v3 like API, then we need to assume where the center is (center determines which rank given subrange belongs to).

`sliding_view(dr, 5)` -> center seems natural to be the 3th element
`sliding_view(dr,6)` -> center is the 3rd or 4th element?

What about case when `halo.prev != halo.next`, 
e.g. halo.prev=1, halo.next=10
where should be center of `sliding_view(dr, 5)`? natural 3rd element is incorrect

We could let user to specify center but then we end up with incompatible API - same situation as now.

We can remove `prev!=next` halo (halo can have one radius only)

We can rename sliding_view to halo_view.

These were my dilemmas. No solution seems to be perfect so I selected the simplest one. What do you propose?